src/templates/base.html

```diff
@@ -13,13 +13,32 @@
     {% block head %}
       <title>{% block title %}The Web Almanac{% endblock %}</title>
       <script async src="https://www.googletagmanager.com/gtag/js?id=UA-22381566-3" nonce="{{ csp_nonce() }}"></script>
+      <script async src="/static/js/web-vitals.js" ></script>
       <script nonce="{{ csp_nonce() }}">
           window.dataLayer = window.dataLayer || [];
           function gtag() {
               dataLayer.push(arguments);
           }
           gtag('js', new Date());
           gtag('config', 'UA-22381566-3');
+
+          function sendWebVitalsGAEvents(metric) {
+            gtag('event', metric.name, {
+              event_category: 'Web Vitals',
+              value: Math.round(metric.name === 'CLS' ? metric.delta * 1000 : metric.delta),
+              event_label: metric.id,
+              non_interaction: true,
+            });
+          }
+
+          window.addEventListener('load', function() {
+            webVitals.getFCP(sendWebVitalsGAEvents);
+            webVitals.getLCP(sendWebVitalsGAEvents, true);
+            webVitals.getCLS(sendWebVitalsGAEvents, true);
```

Comment:
I thought that it would benefit in having overview over frequency:
_Total events per Event Label and Page_
and total value:
_Event Value per Event Label and Page_

Of course it comes with a cost of sending a sequence of same metric event requests (I've seen up to 3 events of same type locally).

If we set reportAllChanges=false, then the event is sent before navigating away, once per page:
_Event Value per Event Label and Page_
OR cumullative over all pageviews:
_Event Value per Page_

Do you think frequency will make sense?

P.S. Hope you'll share collected insights.

Reply:
I also saw some LCP events that happen after rendering is finished (fires together with FID) may have 0 delta.
Don't know if it gives you some performance issue hint, but I'd like your feedback if it makes sense to collect such events.

Reply:
I'd prefer to have one metric reported per page load to keep the analytics simple. Reporting non-final values isn't recommended as @bazzadp pointed out and I think this means we will need to report the `value` rather than the `delta`.

cc @philipwalton

Reply:
Got you.
But I'd still prefer to keep `metric.delta` as a event value. It's equal to `metric.value` in the first event, but in case there are several events sent - aggregated `metric.delta` will report correct value, but aggregated `metric.value` will inflate total event value in report.
And there will be no need to change it, if you decide to collect all events in future.

Reply:
Final changes LGTM!

> I'd prefer to have one metric reported per page load to keep the analytics simple. Reporting non-final values isn't recommended as @bazzadp pointed out

@rviscomi FYI, CLS is the exception to this, see this explanation in the README:

> _**Important:** unlike other metrics, CLS continues to monitor changes for the entire lifespan of the page&mdash;including if the user returns to the page after it's been hidden/backgrounded. However, since browsers often [will not fire additional callbacks once the user has backgrounded a page](https://developers.google.com/web/updates/2018/07/page-lifecycle-api#advice-hidden), `onReport` is always called when the page's visibility state changes to hidden. As a result, the `onReport` function might be called multiple times during the same page load (see [Reporting only the delta of changes](#report-only-the-delta-of-changes) for how to manage this)._

Reply:
Gotcha, so we'd have to aggregate event values by SUM when looking at session-level stats.

Reply:
Another thing to consider is that GA has a 500 hit per session limit. Firing a lot of events on a single page can quickly get you to that limit.

Reply:
True, if you're using the `reportAllChanges` feature and sending a hit to GA with every callback. If you're not using `reportAllChanges` then the max number of hits you will send is three plus the number of times the tab is backgrounded in the same session.

I just did a quick check on my site, and over the last 28 days, the 99th percentile of "Web Vitals" events in a single session is 5 (the 99.9th percentile is 11), so probably nothing to worry about.